otherlibs/stdune/src/copyfile_stubs.c

```diff
@@ -50,13 +50,66 @@ CAMLprim value stdune_copyfile(value v_from, value v_to) {
   CAMLreturn(Val_unit);
 }
 
+CAMLprim value stdune_sendfile(value v_out, value v_in, value v_size) {
+  caml_failwith("sendfile: linux");
+}
+
 CAMLprim value stdune_is_darwin(value v_unit) {
   CAMLparam1(v_unit);
   CAMLreturn(Val_true);
 }
 
+CAMLprim value stdune_is_linux(value v_unit) {
+  CAMLparam1(v_unit);
+  CAMLreturn(Val_false);
+}
+
+#elif __linux__
+
+#include <caml/threads.h>
+#include <caml/unixsupport.h>
+
+#include <sys/sendfile.h>
+
+#define FD_val(value) Int_val(value)
+
+CAMLprim value stdune_copyfile(value v_from, value v_to) {
+  caml_failwith("copyfile: only on macos");
+}
+
+CAMLprim value stdune_sendfile(value v_out, value v_in, value v_size) {
```

Comment:
in the interest of readability, I'd change the argument order here to (src, dst) rather than (dst, src).